addons/mail/static/src/core/common/record.js

```diff
@@ -63,7 +63,7 @@ function updateAttr(record, fieldName, value) {
         typeof value === "string" &&
         !(value instanceof Markup)
```

Comment:
It's also missing an "else" case for this one actually, if the typeof value is is markup, we need to compare values with toString (writing the same markup on top of existing markup should not do the write again, the condition comparing reference in the else is not enough)

Reply:
So we have 4 cases to handle:
- standard (nothing is markup, or not flagged): compare values
- existing is markup, new one is not: replace
- existing is not markup, new one is: replace
- both are markup: compare values (this is the missing case)

Reply:
Inner branchings could use same LOCs, and cast value to assign depending on these conditions